core/src/main/kotlin/org/evomaster/core/problem/external/service/ExternalServiceHandler.kt

```diff
@@ -145,14 +155,22 @@ class ExternalServices {
         return wm
     }
 
+    private fun stopWireMockServers() {
+        externalServices.forEach {
+            it.stopWireMockServer()
+        }
+    }
+
     /**
      * Update the DNS cache with a different IP address for a given host to enable spoofing.
      * If there is an entry already it'll skip from adding to the cache.
      */
     private fun updateDNSCache(host : String, address : String) {
         val entry = DnsCacheManipulator.getDnsCache(host)
-        if (entry != null) {
+        if (entry == null) {
             DnsCacheManipulator.setDnsCache(host, address)
+        } else {
+            log.warn("$host already has a DNS cache entry.")
```

Comment:
naming is bit confusing, ie, `update` that gives a warning if already set. 
rather call it `bindDNSCache`, and return a boolean (no warn) to see if bound or not (ie already present)